### PR TITLE
Issue #7768: Update AbstractChecks to log DetailAST - AnnotationLocation

### LINKED ISSUE #7768: Update AbstractChecks to log DetailAST - AnnotationLocation
Child of https://github.com/checkstyle/checkstyle/issues/5777

This issue specifically focuses on AnnotationLocation .

### PR DESCRIPTION
**Pull request by Malintha1996**
This resolves #7768 

## site update:
Capture

## diff file:
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/index.html
## Goal 
Update AbstractChecks to log DetailAST - AnnotationLocation

### CODE REVIEW

**Comment by Malintha1996** [comment]
> > diff file:
> 
> Waiting on difference file.

done

**Comment by rnveach** [comment]
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/android-launcher/index.html
Configuration was not modified for the check you are working on. ThrowsCount is not the module you are working on. You must change config.

We are expecting differences in regression for this PR.

**Comment by Malintha1996** [comment]
> https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/android-launcher/index.html
> Configuration was not modified for the check you are working on. ThrowsCount is not the module you are working on. You must change config.
> 
> We are expecting differences in regression for this PR.

done

**Comment by rnveach** [comment]
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/checkstyle/index.html
> Number of unique base messages reported below: 115
Number of unique patch messages reported below: 121

Counts don't match up as was mentioned in the parent issue for this.

Please find the new violations that weren't being reported before and ensure we have them as directly part of our `AnnotationLocation` tests.

**Comment by Malintha1996** [comment]
> https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/checkstyle/index.html
> 
> > Number of unique base messages reported below: 115
> > Number of unique patch messages reported below: 121
> 
> Counts don't match up as was mentioned in the parent issue for this.
> 
> Please find the new violations that weren't being reported before and ensure we have them as directly part of our `AnnotationLocation` tests.

The new violations are generated as annotations on the same line has been considered as different contexts(entities) in the AST. For example, when we have @Annotation1 @Annotation2 two messages for Annotation1 and Annotation 2 are created. But before, only one violation message was generated. I have already added this new violation messages to the assertions in the test cases.

**Comment by rnveach** [comment]
>  I have already added this new violation messages to the assertion lists in the test cases.

Can you point me to which ones, I wasn't seeing them when I looked.

**Comment by Malintha1996** [comment]
> > I have already added this new violation messages to the assertion lists in the test cases.
> 
> Can you point me to which ones, I wasn't seeing them when I looked.



> > I have already added this new violation messages to the assertion lists in the test cases.
> 
> Can you point me to which ones, I wasn't seeing them when I looked.

#158,#159  #A161, #A162   #170,#171  #A173,#A174   #182,#183    #A185,#A186

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheckTest.java [hunk 1]
```diff
@@ -285,10 +285,13 @@ public void testAnnotationParameterized() throws Exception {
         checkConfig.addAttribute("allowSamelineSingleParameterlessAnnotation", "false");
         checkConfig.addAttribute("allowSamelineParameterizedAnnotation", "true");
         final String[] expected = {
-            "15: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
-            "17: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
-            "23: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
-            "25: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "15:5: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "17:5: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "17:17: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
```
Comment (Malintha1996) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheckTest.java]: @rnveach

Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheckTest.java]: Thanks. This is what I was looking for.

**Pull request merged**